src/pages/organisationUnits/form/GeometryFields.tsx

```diff
@@ -0,0 +1,75 @@
+import i18n from '@dhis2/d2-i18n'
+import { Field, InputField } from '@dhis2/ui'
+import React from 'react'
+import { useField } from 'react-final-form'
+
+export function GeometryFields() {
+    const fieldName = 'geometry'
+    const { input, meta } = useField(fieldName)
+
+    const handleChange = ({
+        longitude,
+        latitude,
+    }: {
+        longitude?: string
+        latitude?: string
+    }) => {
+        const geometry = {
+            type: 'Point',
+            coordinates: [longitude || undefined, latitude || undefined],
+        }
+
+        input.onChange(geometry)
+    }
+
+    return !input.value || input.value?.type === 'Point' ? (
+        <>
+            <Field
+                name={fieldName}
+                error={meta.touched && meta.error}
+                validationText={meta.touched ? meta.error : undefined}
+            >
+                <InputField
+                    onChange={(e) =>
+                        handleChange({
+                            longitude: e.value,
+                            latitude: input.value.coordinates?.[1],
+                        })
+                    }
+                    label={i18n.t('Longitude')}
+                    inputWidth="400px"
+                    name="longitude"
+                    type="number"
+                    value={input.value.coordinates?.[0].toString()}
+                    min="-90"
+                    max="90"
+                    step="any"
+                />
+                <InputField
+                    onChange={(e) =>
+                        handleChange({
+                            longitude: input.value.coordinates?.[0],
+                            latitude: e.value,
+                        })
+                    }
+                    inputWidth="400px"
+                    label={i18n.t('Longitude')}
+                    name="latitude"
+                    type="number"
+                    value={input.value.coordinates?.[1].toString()}
+                    min="-180"
+                    max="180"
+                    step="any"
+                />
+            </Field>
+        </>
+    ) : (
+        <InputField
+            value={i18n.t('{{type}} coordinates are not editable', {
```

Comment:
We should probably translate this by adding by `geometryType` object to `translatedModelConstants`, and use `type: getConstantTranslation: input.value?.type`